test/Gazel.Tutorial.Test.ProductManagement/AddRemoveProduct.cs

```diff
@@ -40,8 +40,8 @@ public void GIVEN_there_exists_a_product_and_a_cart__WHEN_user_specifies_amount_
         [Test]
         public void GIVEN_there_exists_a_product_in_cart__WHEN_user_removes_product_from_cart__THEN_product_is_removed_from_cart()
         {
-            var removedProduct = CreateProduct();
-            var leftProduct = CreateProduct();
+            var removedProduct = CreateProduct(name: "removed");
+            var leftProduct = CreateProduct(name: "left");
```

Comment:
bak burada daha akilli davranabilirsin, bu case daha once urun adina depend etmiyordu. urun adi ile ilgili constraint gelince niye etsin ki...

CreateProduct bunun icin var zaten, CreateProduct'a isim verilmezse, random generate etsin, o zaman test'leri duzenlemene gerek kalmaz

Reply:
CreateProduct içinde yapamadım, CreateProduct'a isim verilmediğinde isimsiz ürün oluşturamama kontrolünü geçemiyor, CreateRandomProduct diye yeni bir helper methodu yazdım, product için isim, fiyat ve stok bilgisinin gereksiz olduğu testlerde ve cart içinde ürün oluştururken CreateRandomProduct kullandım